### DISCUSSION TITLE
Clear cached credentials in IntelliJ IDEA

### DISCUSSION THREAD

**Discussion by saro1993** [community/community#140527] [Apps, API and Webhooks] [answered]
### Select Topic Area

Question

### Body

How can I programmatically clear cached credentials in IntelliJ IDEA using JetBrains APIs or automate it with scripts across multiple machines?

**Comment by EfremMic** [comment, marked as answer]
To clear cached credentials programmatically in IntelliJ IDEA, you can use JetBrains’ OpenAPI to manage stored credentials. Specifically, the PasswordSafe API allows accessing and removing stored passwords.

Example in Kotlin:
import com.intellij.credentialStore.PasswordSafe

PasswordSafe.getInstance().set(null, null)

To automate clearing across machines:

Use shell scripts to delete cached credentials in system directories (Linux: ~/.config/JetBrains/, Windows: C:\Users\...\AppData\Roaming\JetBrains\).
Automate IntelliJ plugin deployment with Gradle for consistent configuration across machines.
This ensures the cache is cleared effectively across environments.
